docs/source/examples/molecular_hydrogen.myst

```diff
@@ -11,7 +11,7 @@ kernelspec:
   name: python3
 ---
 
-# Estimating the potential energy surface of molecular Hydrogen with ZNE
+# Estimating the potential energy surface of molecular Hydrogen with ZNE and PennyLane
```

Comment:
At some point, this file should be renamed as `molecular_hydrogen_pennylane.py` since our scope is not to replace the existing example but to add a new one.